src/frontend/src/flows/manage/deviceSettings.ts

```diff
@@ -122,37 +105,43 @@ export const resetPhrase = async ({
       )
     : connection;
   if (opConnection === null) {
-    // User aborted, just return
-    reload();
-    return;
+    // User aborted
+    return reload();
   }
 
-  // Save the old pubkey (used as index for replace)
-  const oldKey = device.pubkey;
-  device.pubkey = Array.from(
-    new Uint8Array(recoverIdentity.getPublicKey().toDer())
-  );
-
-  try {
-    const phrase = userNumber.toString(10) + " " + recoveryPhrase;
+  const uploadPhrase = (pubkey: DerEncodedPublicKey): Promise<void> =>
+    withLoader(() =>
+      opConnection.replace(device.pubkey, {
+        ...device,
+        pubkey: Array.from(new Uint8Array(pubkey)),
+      })
+    );
 
-    const res = await displayAndConfirmPhrase({ phrase, operation: "reset" });
+  const res = await phraseWizard({
+    userNumber,
+    operation: "reset",
+    uploadPhrase,
+  });
 
-    if (res === "confirmed") {
-      await withLoader(() => opConnection.replace(oldKey, device));
-    } else if (res !== "canceled") {
-      unreachableLax(res);
-    }
-  } catch (e: unknown) {
+  if (typeof res === "object" && "ok" in res) {
+    // If the user was authenticated with the phrase, then replace the connection
+    // to use the new phrase to void logging them out
+    const nextConnection = sameDevice
+      ? await connection.fromIdentity(userNumber, res.ok)
+      : undefined;
+    return reload(nextConnection);
+  } else if (typeof res === "object" && "error" in res) {
     await displayError({
       title: "Could not reset recovery phrase",
       message: "An unexpected error occurred",
-      detail: e instanceof Error ? e.toString() : "unknown error",
+      detail: unknownToString(res.error, "unknown error"),
       primaryButton: "Ok",
     });
+    return reload();
```

Comment:
nth: you can spare on


```
  } else if (typeof res === "object" && "error" in res) {
     ...
  } else {
    assertType<"canceled">(res);
  }

  return reload();
}

Reply:
true, that's kind of what the original code was doing, although I found it less clear/explicit. I don't mind the extra line but I can move it if you feel strongly about it!